src/test/java/com/shima/todo_list/service/TodoListServiceTest.java

```diff
@@ -93,4 +93,25 @@ public class TodoListServiceTest {
                 });
         verify(todoListMapper).findById(99);
     }
+
+    //DELETE機能のテスト
+    @Test
+    public void 指定したIDのデータが削除できること() {
+        doReturn(Optional.of(new Todo(2, "API作成", LocalDate.of(2023, 12, 7), null, null))).when(todoListMapper).findById(2);
+        todoListService.findById(2);
+        verify(todoListMapper).findById(2);
+        verify(todoListMapper).findById(2);
+
+    }
+
+    @Test
+    public void 存在しないIDのデータを削除したときに例外処理が動作こと() {
+        doReturn(Optional.empty()).when(todoListMapper).findById(99);
+
+        assertThatThrownBy(
+                () -> todoListService.findById(99)
+        ).isInstanceOfSatisfying(
+                TaskNotFoundException.class, e -> assertThat(e.getMessage()).isEqualTo("task not found"));
+        verify(todoListMapper).findById(99);
```

Comment:
こちら、deleteが呼ばれていないことも確認すると、より丁寧ですね！
```suggestion
        verify(todoListMapper).findById(99);
        verify(todoListMapper, never()). delete(any()); 
```

Reply:
こうゆう認識でしょうか。
delete(any()　→  削除(任意())

4333cec5081eee0cafb5db5e61bf4ff7122d21e5

Reply:
このテストもfindByIdを実行するテストになってしまっているので、deleteを実行するテストに修正しましょう。

> delete(any()　→ 削除(任意())

その認識であっています！
が、「verifyとは何なのか？」について、もう一度調べて理解を深めておいた方が良いと思います！

Reply:
verifyについて意味 = 確認する。
`verify(モックインスタンス, times(想定回数)).メソッド(引数);`
モック化したメソッドの呼び出し回数を検証する(verify)
`doNothing()`や`verify()`両方行う必要はなく、`verify()`のみで2つ同時に行うことができる。


726f8a5c425a690fc55a1acd5b17030807f6e90b